ui/src/app/components/prepare-migration/prepare-migration.component.ts

```diff
@@ -437,10 +443,14 @@ export class PrepareMigrationComponent implements OnInit {
       this.dataflowConfig = {
         network: localStorage.getItem(Dataflow.Network) as string,
         subnetwork: localStorage.getItem(Dataflow.Subnetwork) as string,
-        hostProjectId: localStorage.getItem(Dataflow.HostProjectId) as string
+        hostProjectId: localStorage.getItem(Dataflow.HostProjectId) as string,
+        maxWorkers: localStorage.getItem(Dataflow.MaxWorkers) as string,
+        numWorkers: localStorage.getItem(Dataflow.NumWorkers) as string,
+        serviceAccountEmail: localStorage.getItem(Dataflow.ServiceAccountEmail) as string
       }
+      console.log(this.dataflowConfig)
       this.isDataflowConfigurationSet = localStorage.getItem(Dataflow.IsDataflowConfigSet) as string === 'true'
-      if (this.isSharded) {
+      if (this.isDataflowConfigurationSet) {
```

Comment:
@manitgupta switched this if condition to handle both sharded and non sharded flow, any thing that I could be missing?

Reply:
This doesn't look right since `setDataflowDetailsForShardedMigrations` call would be made for non-sharded flow as well with this case.

Reply:
Right, reverted.